api/v1beta1/azurecluster_webhook_test.go

```diff
@@ -220,36 +220,35 @@ func TestAzureCluster_ValidateUpdate(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "azurecluster resource group is immutable",
-			oldCluster: &AzureCluster{
-				Spec: AzureClusterSpec{
-					ResourceGroup: "demoResourceGroup",
-				},
-			},
-			cluster: &AzureCluster{
-				Spec: AzureClusterSpec{
-					ResourceGroup: "demoResourceGroup-2",
-				},
-			},
+			name: "azurecluster subscription ID is immutable without annotation",
+			oldCluster: func() *AzureCluster {
+				ac := createValidCluster()
+				ac.Spec.SubscriptionID = "212ec1q8"
+				return ac
+			}(),
+			cluster: func() *AzureCluster {
+				ac := createValidCluster()
+				ac.Spec.SubscriptionID = "212ec1q9"
+				return ac
+			}(),
```

Comment:
I noticed this test did not create a valid cluster, which is bad, because that means it can return a false negative. I changed it to create a valid cluster.

Some other tests in this file have the same issue. I can fix these in a separate PR.